### PR TITLE
Scaling factor governor

### PR DESCRIPTION
**Pull request by vuong177**
Closes: #1460

## What is the purpose of the change

> Add a description of the overall background and high level changes that this PR introduces

*(E.g.: This pull request improves documation of area A by adding ....*

## Brief Changelog

*(for example:)*
 
  - *The metadata is stored in the blob store on job creation time as a persistent artifact*
  - *Deployments RPC transmits only the blob storage reference*
  - *Daemons retrieve the RPC data from the blob cache*

## Testing and Verifying

*(Please pick one of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added unit test that validates ...*
  - *Added integration tests for end-to-end deployment with ...*
  - *Extended integration test for ...*
  - *Manually verified the change by ...*

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: x/gamm/keeper/msg_server.go [hunk 1]
```diff
@@ -280,3 +280,13 @@ func (server msgServer) ExitSwapShareAmountIn(goCtx context.Context, msg *types.
...
+
+// func (server msgServer) StableSwapAdjustScalingFactors(goCtx context.Context, msg *stableswap.MsgStableSwapAdjustScalingFactors) (*stableswap.MsgStableSwapAdjustScalingFactorsResponse, error) {
+// 	ctx := sdk.UnwrapSDKContext(goCtx)
+
+// 	if err := server.keeper.setStableSwapScalingFactors(ctx, msg.ScalingFactors, msg.PoolID, msg.ScalingFactorGovernor); err != nil {
+// 		return nil, err
+// 	}
+
+// 	return &stableswap.MsgStableSwapAdjustScalingFactorsResponse{}, nil
+// }
```
Comment (ValarDragon) [inline x/gamm/keeper/msg_server.go:292]: Why's this commented out?

File: proto/osmosis/gamm/pool-models/stableswap/tx.proto [hunk 2]
```diff
@@ -29,3 +31,16 @@ message MsgCreateStableswapPool {
 message MsgCreateStableswapPoolResponse {
   uint64 pool_id = 1 [ (gogoproto.customname) = "PoolID" ];
 }
+
+message MsgStableSwapAdjustScalingFactors {
+  string scaling_factor_governor = 1
+      [ (gogoproto.moretags) = "yaml:\"scaling_factor_governor\"" ];
```
Comment (ValarDragon) [inline proto/osmosis/gamm/pool-models/stableswap/tx.proto:37]:
```suggestion
  // Sender must be the pool's scaling_factor_governor in order for the tx to succeed
  string sender = 1
      [ (gogoproto.moretags) = "yaml:\"sender\"" ];
```

**Review by ValarDragon — COMMENTED** [review]
Awesome, thanks so much for adding this!

Want to just add a test that ensures the factors do get updated in state, then we can merge it in?

File: x/gamm/keeper/msg_server.go (hunk 1 above)
Comment (vuong177) [reply x/gamm/keeper/msg_server.go:292]: Because stableswap msg server still disable in main branch. I think better if we comment out that code !

Comment (ValarDragon) [reply x/gamm/keeper/msg_server.go:292]: Fair enough

**Comment by vuong177** [comment]
I added test but commented it out because stableswap still disable

File: x/gamm/keeper/msg_server.go [hunk 3]
```diff
@@ -280,3 +280,13 @@ func (server msgServer) ExitSwapShareAmountIn(goCtx context.Context, msg *types.
...
 }
+
+// func (server msgServer) StableSwapAdjustScalingFactors(goCtx context.Context, msg *stableswap.MsgStableSwapAdjustScalingFactors) (*stableswap.MsgStableSwapAdjustScalingFactorsResponse, error) {
+// 	ctx := sdk.UnwrapSDKContext(goCtx)
+
+// 	if err := server.keeper.setStableSwapScalingFactors(ctx, msg.ScalingFactors, msg.PoolID, msg.ScalingFactorGovernor); err != nil {
+// 		return nil, err
+// 	}
+
+// 	return &stableswap.MsgStableSwapAdjustScalingFactorsResponse{}, nil
```
Comment (mattverse) [inline x/gamm/keeper/msg_server.go:291]: `stableswap` -> `types`

Comment (vuong177) [reply x/gamm/keeper/msg_server.go]: Is this necessary ? I see we use `stableswap` in function `CreateStableswapPool` (line 51).

Comment (mattverse) [reply x/gamm/keeper/msg_server.go]: yeah my bad it looks fine 👍

**Review by mattverse — APPROVED** [review]
LGTM

**Pull request merged**